LabExT/Movement/PathPlanning.py

```diff
@@ -126,3 +129,158 @@ def _create_outline(
             y_max += self.FIBER_LENGTH
 
         return x_min, x_max, y_min, y_max
+
+
+class PotentialField:
+    """
+    Waypoint calculation with potential field algorithm.
+    A potential field is created for each stage, with all other stages being obstacles.
+
+    This algorithm operates in 2D (x,y), assuming that the stages are on a plane.
+    """
+
+    REPULSIVE_GAIN = 100.0
+    ATTRACTIVE_GAIN = 5.0
```

Comment:
Probably needs to be adaptable eventually. In my simulations, I found that Attractive gain should be around 1, repulsive gain around 10000.

Reply:
These are only parameters of the constructor